project/packages/dtos/src/users.ts

```diff
@@ -21,21 +23,28 @@ export const userCollectionSchema = z.object({
   users: z.array(userSchema),
 });
 
+export const sortUsersQuerySchema = z.object({
+  field: z.string(),
+  order: z.enum(["asc", "desc"]),
+});
+
 export const userFiltersSchema = z.object({
-  email: emailSchema,
-  username: z.string().optional(),
+  email: emailSchema.optional(),
+  username: usernameSchema.optional(),
```

Comment:
Using the usernameSchema here causes the filter in the table to make a false check on whether there are <4 characters when the user is filtering

Reply:
updated!